src/tools/rustdoc-gui/tester.js

```diff
@@ -213,9 +208,12 @@ async function main(argv) {
         );
         if (no_headless) {
             await tests[i];
+        } else if (opts["jobs"] > 0 && tests.length >= opts["jobs"]) {
+            await Promise.all(tests);
```

Comment:
```suggestion
            await Promise.race(tests);
```

This should start a new test as soon as one finishes instead of waiting for all tests in a batch to finish, which may not happen at the same time.

Reply:
It would be a nice improvement indeed!